Lib/test/test_re.py

```diff
@@ -599,8 +603,83 @@ def test_match_getitem(self):
         with self.assertRaises(TypeError):
             m[0] = 1
 
-        # No len().
-        self.assertRaises(TypeError, len, m)
+    def test_match_sequence(self):
```

Comment:
Let's split this into multiple test: `_index`, `_count` and `_iter`, as well as `match case`. For `getitem`, let's move them under `test_match_getitem` as well and reduce the number of duplicated checks.